### PR TITLE
Gml 1809 feedback analysis access control

### PR DESCRIPTION
**Pull request by luzhoutg**
1. update feedback analytics notebook with get_feedback endpoint
2. add access control
3. add test cases
4. update the loadConfig to load multiple config files

### CODE REVIEW

**Review by parkererickson-tg — APPROVED** [review]

File: chat-history/routes/routes.go [hunk 1]
```diff
@@ -154,3 +156,152 @@ func auth(userId string, r *http.Request) (string, int, []byte, bool) {
...
 }
+
+// executeGSQL sends a GSQL query to TigerGraph with basic authentication and returns the response
+func executeGSQL(host, username, password, query string, tgcloud bool) (string, error) {
+	var requestURL string
+	// Construct the URL for the GSQL query endpoint
+	if tgcloud {
+		requestURL = fmt.Sprintf("%s:443/gsqlserver/gsql/file", host)
+	} else {
+		requestURL = fmt.Sprintf("%s:14240/gsqlserver/gsql/file", host)
```
Comment (billshitg) [inline chat-history/routes/routes.go:167]: For on-prem DBs, the port is not always 14240. Could we read it from config? How are we handling it for copilot?

Comment (luzhoutg) [reply chat-history/routes/routes.go]: Thanks for pointing this out. Just push the updated. For CoPilot, pyTG have handled it already.

File: chat-history/config/config.go [hunk 2]
```diff
@@ -29,25 +32,28 @@ type Config struct {
...
+			if err != nil {
+				return Config{}, err
+			}
 		}
-		b = []byte(cfg)
-	} else {
-		b, err = os.ReadFile(path)
-		if err != nil {
+
+		if err := json.Unmarshal(b, &cfg); err != nil {
```
Comment (RobRossmiller-TG) [inline chat-history/config/config.go:54]: If you're reading multiple config files, won't cfg just be the last file that was parsed in the list?

**Review by RobRossmiller-TG — CHANGES_REQUESTED** [review]
- Tests could do with a negative test (i.e. user is only allowed to see their messages, not all)
- One quirk in LoadConfig

Otherwise, looks good!

File: chat-history/config/config.go (hunk 2 above)
Comment (luzhoutg) [reply chat-history/config/config.go]: Good catch. I update the loadConfig to load multiple config file in to different config structs. i.e., ChatDbConfig, and TgDbConfig. It looks like this: `	router.HandleFunc("GET /get_feedback", routes.GetFeedback(cfg.TgDbConfig.Hostname, cfg.TgDbConfig.GsPort, cfg.ChatDbConfig.ConversationAccessRoles, cfg.TgDbConfig.TgCloud))
`

Comment (luzhoutg) [reply chat-history/config/config.go]: also added test cases for non-admin users and non-existent users.

**Review by RobRossmiller-TG — APPROVED** [review]

**Pull request merged**